cpp/src/arrow/io/interfaces.h

```diff
@@ -24,6 +24,7 @@
 
 #include "arrow/io/type_fwd.h"
 #include "arrow/type_fwd.h"
+#include "arrow/util/async_generator.h"
```

Comment:
This is a too heavy inclusion, should instead just use the corresponding `std::function<...>` definition.

Reply:
Not in this `.h`, only in the `.cc`

Reply:
Ah I see